src/openms/source/FORMAT/HANDLERS/UnimodXMLHandler.cpp

```diff
@@ -102,44 +102,33 @@ namespace OpenMS
         {
           position = ResidueModification::ANYWHERE;
         }
+        else if (pos == "Protein N-term")
+        {
+          position = ResidueModification::N_TERM;
+        }
+        else if (pos == "Protein C-term")
+        {
+          position = ResidueModification::C_TERM;
+        }
+        else if (pos == "Any C-term")
+        {
+          position = ResidueModification::C_TERM;
+        }
+        else if (pos == "Any N-term")
+        {
+          position = ResidueModification::N_TERM;
+        }
         else
         {
-          if (pos == "Protein N-term")
-          {
-            position = ResidueModification::N_TERM;
-          }
-          else
-          {
-            if (pos == "Protein C-term")
-            {
-              position = ResidueModification::C_TERM;
-            }
-            else
-            {
-              if (pos == "Any C-term")
-              {
-                position = ResidueModification::C_TERM;
-              }
-              else
-              {
-                if (pos == "Any N-term")
-                {
-                  position = ResidueModification::N_TERM;
-                }
-                else
-                {
-                  warning(LOAD, String("Don't know allowed position called: '") + pos  + "' - setting to anywhere");
-                }
-              }
-            }
-          }
+          warning(LOAD, String("Don't know allowed position called: '") + pos  + "' - setting to anywhere");
         }
 
         if (!pos.hasSubstring("Protein"))
         {
           was_valid_peptide_modification_ = true;
           term_specs_.push_back(position);
-          sites_.push_back(site);
+          if (site.size() > 1) site = "X"; // C-term/N-term
```

Comment:
I was about to agree, but since I use `site[0]` in the next line, I think it's better to treat anything unexpected as "X". Otherwise I'd have to check for things that are neither residues nor "C/N-term" and throw an error.